runtime/ops/tty.rs

```diff
@@ -43,6 +43,27 @@ pub fn init() -> Extension {
     .build()
 }
 
+#[cfg(windows)]
+const COOKED_MODE: DWORD = // ref: <https://learn.microsoft.com/en-us/windows/console/setconsolemode> @@ <https://archive.is/HWHwG>
+  wincon::ENABLE_LINE_INPUT // enable line-by-line input (returns input only after CR is read)
+  | wincon::ENABLE_ECHO_INPUT // enables real-time character echo to console display (requires ENABLE_LINE_INPUT)
+  | wincon::ENABLE_PROCESSED_INPUT // system handles CTRL-C (with ENABLE_LINE_INPUT, also handles BS, CR, and LF) and other control keys (when using `ReadFile` or `ReadConsole`)
+  ;
+
+#[cfg(windows)]
+fn mode_raw_input_on(original_mode: DWORD) -> DWORD {
+  original_mode // disable, then enable (`&` has higher precedence than `|`)
+    & ! COOKED_MODE // disable line-by-line input, character echo, and system control key handling
+    | wincon::ENABLE_VIRTUAL_TERMINAL_INPUT // enable console/system handling of virtual terminal sequences
+}
+
+#[cfg(windows)]
+fn mode_raw_input_off(original_mode: DWORD) -> DWORD {
+  original_mode // disable, then enable (`&` has higher precedence than `|`)
+    & ! wincon::ENABLE_VIRTUAL_TERMINAL_INPUT // disable console/system handling of virtual terminal sequences
+    | COOKED_MODE // enable line-by-line input, character echo, and system control key handling
+}
```

Comment:
Less verbose. Also, don't think we need an archive link because that can always be found via the original url.

```suggestion
// ref: https://learn.microsoft.com/en-us/windows/console/setconsolemode
#[cfg(windows)]
const COOKED_MODE: DWORD =
  // line-by-line input (returns input only after CR is read)
  wincon::ENABLE_LINE_INPUT
  // real-time character echo to console display (requires ENABLE_LINE_INPUT)
  | wincon::ENABLE_ECHO_INPUT
  // system handles CTRL-C (with ENABLE_LINE_INPUT, also handles BS, CR, and LF)
  // and other control keys (when using `ReadFile` or `ReadConsole`)
  | wincon::ENABLE_PROCESSED_INPUT;

#[cfg(windows)]
fn mode_raw_input_on(original_mode: DWORD) -> DWORD {
  original_mode & !COOKED_MODE | wincon::ENABLE_VIRTUAL_TERMINAL_INPUT
}

#[cfg(windows)]
fn mode_raw_input_off(original_mode: DWORD) -> DWORD {
  original_mode & !wincon::ENABLE_VIRTUAL_TERMINAL_INPUT | COOKED_MODE
}
```